commands/invoke_test.go

```diff
@@ -0,0 +1,47 @@
+package commands
+
+import (
+	"net/http"
+	"testing"
+
+	"os"
+
+	"regexp"
+
+	"github.com/openfaas/faas-cli/test"
+)
+
+func Test_invoke(t *testing.T) {
+	expectedResponse := "response-test-data"
+	funcName := "test-1"
+	s := test.MockHttpServer(t, []test.Request{
+		{
+			Method:             http.MethodPost,
+			Uri:                "/function/" + funcName,
+			ResponseStatusCode: http.StatusOK,
+			ResponseBody:       expectedResponse,
+		},
+	})
+	defer s.Close()
+
+	os.Stdin, _ = os.OpenFile("./stdin", os.O_CREATE, 0666)
```

Comment:
@alexellis What do you think of this approach

Reply:
Looks like a good idea to me, and the test do pass. However, for me, it looks like the data that is written to the './stdin' file (`test-data`) is not being sent to the mock server. On line 71 is `invoke.go` is the following line:
```
functionInput, err := ioutil.ReadAll(os.Stdin)
```

For me, `functionInput` is an empty byte array after this line

Reply:
on line 42 I added: 	`fmt.Printf("functionInput: " + string(functionInput) + "\n")`
`
	invoke_test.go:42: functionInput: 
		"response-test-data"
`

Reply:
"response-test-data" is the output of the mock server. `functionInput` is assigned that value after the `proxy.InvokeFunction()` is run, and regardless of where you print `functionInput`, it will show the value that `functionInput` has after `proxy.InvokeFunction()` has run (due to `Printf` not executing synchronously). However, if you remove the call to `proxy.InvokeFunction()` and print out `functionInput` you will see that it holds no data:

```
	invoke_test.go:42: functionInput is set to: 
		
```

The issue is that in invoke_test.go, it is set to `'test-data'`:

```
	os.Stdin, _ = os.OpenFile("./stdin", os.O_CREATE, 0666)
	os.Stdin.WriteString("test-data")
	defer func() {
		os.Stdin.Close()
		os.Remove("./stdin")
	}()
```

Reply:
Thanks for pointing out this, I forgot to rewind the pointer in stdin temp file